pkg/istio-agent/agent.go

```diff
@@ -267,8 +267,11 @@ func (a *Agent) generateNodeMetadata() (*model.Node, error) {
 	})
 }
 
-func (a *Agent) initializeEnvoyAgent(ctx context.Context) error {
+func (a *Agent) initializeEnvoyAgent(ctx context.Context, credentialSocketExists bool) error {
```

Comment:
I think the param should be instead a field in agent. It's strange for all other params to be in Agent and this to be passed around as param.